### ISSUE TITLE
incorrect symlink setup for fish's vendor_completions.d

### ISSUE THREAD

**Issue by MerelyAPseudonym** [Homebrew/homebrew#41742]
`$HOMEBREW_PREFIX/share/fish/vendor_completions.d` *should* work like `$HOMEBREW_PREFIX/share/zsh/site-functions`, but it doesn't in at least two ways:

1. If you don't have any third-party `fish` completions currently installed and run `brew install fish`, `$HOMEBREW_PREFIX/share/fish` gets symlinked to `$HOMEBREW_CELLAR/fish/2.2.0/share/fish`:
   ```
   $ ls -l /usr/local/share/fish
   lrwxr-xr-x  1 jtilles  admin  31 Jul 15 09:44 /usr/local/share/fish -> ../Cellar/fish/2.2.0/share/fish
   ```

2. If the `fish` formula is *not* installed and you install (for example) `pass` (making sure to pass `--build-from-source` so that `pass`’s `fish`-completion scripts get installed), you'll see some nonsense like this:
   ```
   $ ls -l /usr/local/share/fish
   lrwxr-xr-x   1 jtilles  admin    31 Jul 15 09:32 /usr/local/share/fish -> ../Cellar/pass/1.6.5/share/fish
   
   $ tree /usr/local/opt/pass/share/fish
   /usr/local/opt/pass/share/fish
   └── vendor_completions.d
       └── pass.fish

   1 directory, 1 file
   ```

In contrast to the two borked examples above, here's what (I think) a "proper" layout looks like.
```
$ ll /usr/local/share/fish
total 40
lrwxr-xr-x  1 josh  admin    46B Jul 13 12:41 completions -> ../../Cellar/fish/2.2.0/share/fish/completions
lrwxr-xr-x  1 josh  admin    46B Jul 13 12:41 config.fish -> ../../Cellar/fish/2.2.0/share/fish/config.fish
lrwxr-xr-x  1 josh  admin    44B Jul 13 12:41 functions -> ../../Cellar/fish/2.2.0/share/fish/functions
lrwxr-xr-x  1 josh  admin    38B Jul 13 12:41 man -> ../../Cellar/fish/2.2.0/share/fish/man
lrwxr-xr-x  1 josh  admin    40B Jul 13 12:41 tools -> ../../Cellar/fish/2.2.0/share/fish/tools
drwxr-xr-x  3 josh  admin   102B Jul 13 12:43 vendor_completions.d

$ ll /usr/local/share/fish/vendor_completions.d
total 8
lrwxr-xr-x  1 josh  admin    68B Jul 13 12:43 pass.fish -> ../../../Cellar/pass/1.6.5/share/fish/vendor_completions.d/pass.fish
```

# TLDR:
These two paths should never be symlinks:
- `$HOMEBREW_PREFIX/share/fish`
- `$HOMEBREW_PREFIX/share/fish/vendor_completions.d`

**Comment by MerelyAPseudonym** [comment]
btw, for all I know `$HOMEBREW_PREFIX/share/fish/{completions,functions,man,tools}` could/should all be proper directories (i.e., not symlinks themselves, but _containing_ symlinks) too

**Comment by MerelyAPseudonym** [comment]
We should keep https://github.com/fish-shell/fish-shell/issues/2113 in mind though. *Perhaps* in the meantime `$HOMEBREW_CELLAR/fish/2.2.0/share/fish/vendor_completions.d` should point to `$HOMEBREW_PREFIX/share/fish/vendor_completions.d`.

**Comment by xu-cheng** [comment]
The related code is [here](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/keg.rb#L276).

Could you try and open a pull request? This [document](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged) should be helpful. Thanks!
